tests/sample/cgroup_mt_connect6.c

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// Whenever this sample program changes, bpf2c_tests will fail unless the
+// expected files in tests\bpf2c_tests\expected are updated. The following
+// script can be used to regenerate the expected files:
+//     generate_expected_bpf2c_output.ps1
+//
+// Usage:
+// .\scripts\generate_expected_bpf2c_output.ps1 <build_output_path>
+// Example:
+// .\scripts\generate_expected_bpf2c_output.ps1 .\x64\Debug\
+
+#include "bpf_endian.h"
+#include "bpf_helpers.h"
+#include "net/ip.h"
+#include "socket_tests_common.h"
+
+const uint16_t remote_port = SOCKET_TEST_PORT;
+const uint16_t redirect_offset = 1000;
+
+SEC("cgroup/connect6")
+int
+tcp_mt_connect6(bpf_sock_addr_t* ctx)
```

Comment:
nit: maybe have both the v4 and v6 ebpf programs in a single file, and then use a common function for both the programs, as most of the below code is same for v4 and v6.

Reply:
This was a deliberate decision.  The original idea was to be able to load 2 different programs (.sys) files just to make sure that there aren't (and a future regression does not cause) any issues loading multiple .sys files, each doing its own thing.   IMO, the fact that both happen to do the same work should be treated as a coincidence.

Granted, we can't do that right now due to the IPv4/IPv6 issue (#2351), but once we fix that, I was planning on enhancing this test to load an IPv4 and and IPv6 program (with some tweaks suggested by Dave and Alan).